wazo_auth/services/saml_config.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022-2024 The Wazo Authors  (see the AUTHORS file)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import logging
+import os
+from xml.etree import ElementTree
+
+from wazo_auth.database.models import Domain, SAMLConfig
+from wazo_auth.plugins.http.saml_config.schemas import saml_config_schema
+from wazo_auth.services.helpers import BaseService
+
+logger = logging.getLogger(__name__)
+
+
+class SAMLConfigService(BaseService):
+    def __init__(self, config, saml_service, dao) -> None:
+        self._xml_files_dir: str = config['saml']['xml_files_dir']
+        self._saml_service = saml_service
+        super().__init__(dao)
+        self._reload_saml_service()
+
+    def _get_metadata_path(self, tenant_uuid: str) -> str:
+        return self._xml_files_dir + '/' + tenant_uuid + '.xml'
+
+    def _update_xml_metadata(
+        self, tenant_uuid: str, etree_metadata: ElementTree.ElementTree
+    ) -> None:
+        etree_metadata.write(self._get_metadata_path(tenant_uuid))
+
+    def _delete_conf(self, tenant_uuid: str) -> None:
+        os.remove(self._xml_files_dir + '/' + tenant_uuid + '.xml')
```

Comment:
Reuse `_get_metadata_path` for the path(avoid duplication of logic, guarantee correctness)
```suggestion
        os.remove(self._get_metadata_path)
```